app/hotels/src/singleHotel/bookNow/BookNow.js

```diff
@@ -82,11 +96,16 @@ export class BookNow extends React.Component<Props> {
           <TouchableItem onPress={this.handleGoToPayment} activeOpacity={0.6}>
             <View style={styles.buttonInnerWrapper}>
               <Text style={styles.buttonTitle}>Book Now</Text>
-              <Price
-                style={styles.buttonPrice}
-                amount={price.amount}
-                currency={price.currency}
-              />
+              <View style={styles.priceWrapper}>
+                <Text style={styles.buttonPrice}>
+                  {this.getBookNowHelpText()}
```

Comment:
I am still not sure about this pattern. I am not sure what is this rerendering behavior here but what do you think just from a design perspective - is it better to call functions like this or to create a component and encapsulate everything like we do everywhere else?

So I am trying to compare this: